src/date-picker/date-picker.scss

```diff
@@ -62,58 +64,251 @@
   display: flex;
-  align-items: center;
-  justify-content: space-evenly;
   padding: 0 $iui-s;
+  margin-bottom: $iui-xxs;
 }
 
-@mixin iui-calendar-day {
-  @include iui-focus;
+@mixin iui-calendar-day-base {
   cursor: pointer;
-  display: inline-block;
-  width: $iui-baseline * 3;
-  line-height: $iui-baseline * 3;
+  text-align: center;
+  width: $iui-date-picker-cell-width;
+  height: $iui-date-picker-cell-height;
   border-radius: $iui-border-radius;
+  font-variant-numeric: tabular-nums;
+}
 
-  &.iui-today {
+@mixin iui-calendar-day-base-hover {
+  &:hover {
     font-weight: $iui-font-weight-semibold;
-    border-radius: 50%;
     @include themed {
-      background-color: rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-6));
+      color: t(iui-color-foreground-primary);
+      background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-6));
+    }
+  }
+}
+
+@mixin iui-calendar-day-today-base {
+  position: relative;
+  font-weight: $iui-font-weight-semibold;
+
+  &::before {
+    content: '';
+    position: absolute;
+    display: block;
+    width: $iui-date-picker-cell-height;
+    height: $iui-date-picker-cell-height;
+    border-radius: $iui-border-radius-round;
+    box-sizing: border-box;
+    top: 50%;
+    left: 50%;
+    transform: translate(-50%, -50%);
+    @include themed {
+      border: 2px solid rgba(t(iui-color-foreground-body-rgb), t(iui-opacity-5));
     }
   }
 
   &:hover {
-    font-weight: $iui-font-weight-semibold;
+    background-color: initial;
     @include themed {
       color: t(iui-color-foreground-primary);
+    }
+  }
+
+  &:hover::before {
+    @include themed {
       background-color: rgba(t(iui-color-foreground-primary-rgb), t(iui-opacity-6));
     }
   }
+}
 
-  &.iui-selected {
-    font-weight: $iui-font-weight-semibold;
-    border-radius: $iui-border-radius;
+@mixin iui-calendar-day-selected-base {
+  font-weight: $iui-font-weight-semibold;
+  @include themed {
+    background-color: t(iui-color-background-primary);
+    color: t(iui-color-foreground-accessory);
+  }
+  @include iui-focus($color: var(--iui-color-foreground-accessory), $offset: -3px);
+}
+
+@mixin iui-calendar-day {
+  &,
+  &-today {
+    @include iui-focus;
+    @include iui-calendar-day-base;
+    @include iui-calendar-day-base-hover;
+  }
+
+  &-today {
+    @include iui-calendar-day-today-base;
+  }
+}
+
+@mixin iui-calendar-day-selected {
+  &,
+  &-today {
+    @include iui-calendar-day-base;
+    @include iui-calendar-day-selected-base;
     cursor: default;
+  }
+
+  &-today {
+    @include iui-calendar-day-today-base;
+
+    &:hover {
+      @include themed {
+        background-color: t(iui-color-background-primary);
+        color: t(iui-color-foreground-accessory);
+      }
+
+      &::before {
+        @include themed {
+          background-color: initial;
+        }
+      }
+    }
+
+    &::before {
+      @include themed {
+        border-color: t(iui-color-foreground-accessory);
```

Comment:
```suggestion
        border-color: rgba(t(iui-color-foreground-accessory-rgb), t(iui-opacity-4));
```
Can you try this?  I think it looks a little less attention grabbing.
![Screen Shot 2022-03-22 at 10 25 36 AM](https://user-images.githubusercontent.com/849817/159504475-a12a86aa-5486-48b8-91bb-dabe4e38cd89.png)